test/pytest/service-desktop/test_contacts.py

```diff
@@ -1,146 +1,357 @@
-            "altName": "Testowy2",
-            "email": "testowy.2@example.com",
-            "blocked": False,
-            "favourite": True,
-            "ice": True,
-            "numbers": ["547623521"],
-            "speedDial": "7",
-            "priName": "Test2",
-            "note": "this is a really cool guy",
-            "id": contact_id_to_update}
-    ret = harness.endpoint_request("contacts", "put", body)
-    assert ret["status"] == status["NoContent"]
-
-    # gathering updated contact
-    body = {"id": contact_id_to_update}
-    ret = harness.endpoint_request("contacts", "get", body)
-    contact = {"address": "6 Czeczota St.\n02600 Warsaw",
-               "altName": "Testowy2",
-               "email": "testowy.2@example.com",
-               "blocked": False,
-               "favourite": True,
-               "ice": True,
-               "numbers": ["547623521"],
-               "speedDial": "7",
-               "priName": "Test2",
-               "note": "this is a really cool guy",
-               "id": contact_id_to_update}
-    assert ret["body"] == contact
-
-    # removing added contact
-    body = {"id": contact_id_to_update}
-    ret = harness.endpoint_request("contacts", "del", body)
-    assert ret["status"] == status["NoContent"]
-
-    # verifying count
-    body = {"count": True}
-    ret = harness.endpoint_request("contacts", "get", body)
-    assert ret["status"] == status["OK"]
-
-    assert ret["body"]["count"] == count
+def test_edit_contact(harness):
+    # Check initial state
+    contacts_tester = ContactsTester(harness)
+    result, initial_number_of_contact_records = contacts_tester.get_contacts_count()
+    assert result, "Failed to get contacts count!"
+
+    # Add contact
+    contact_record = {
+        "address": "6 Czeczota St.\n02600 Warsaw",
+        "altName": "Smith",
+        "email": "john.smith@mudita.com",
+        "blocked": False,
+        "favourite": False,
+        "ice": False,
+        "numbers": [
+            "123456789"
+        ],
+        "speedDial": "1",
+        "priName": "John",
+        "note": "Some note"
+    }
+
+    result, contact_id = contacts_tester.add_contact(contact_record)
+    assert result, "Failed to add contact!"
+
+    # Edit contact
+    edited_contact_record = {
+        "id": contact_id,
+        "address": "6 Czeczota St.\n02600 Warsaw",
+        "altName": "Davies",
+        "email": "john.davies@mudita.com",
+        "blocked": False,
+        "favourite": False,
+        "ice": False,
+        "numbers": [
+            "123456799"
+        ],
+        "speedDial": "1",
+        "priName": "John",
+        "note": "Some note"
+    }
+    assert contacts_tester.update_contact(edited_contact_record), "Failed to update contact!"
+    result, retrieved_contact_record = contacts_tester.get_contact_by_id(contact_id)
+    assert result, "Failed to get contact by ID!"
+    assert retrieved_contact_record == edited_contact_record
```

Comment:
I really like that you add failure messages to your asserts, but why stop here? It would be best to give this one a message too so that you can easily see what went wrong on the report.